src/npcmove.cpp

```diff
@@ -263,6 +318,7 @@ void npc::regen_ai_cache()
     ai_cache.danger = 0.0f;
     ai_cache.total_danger = 0.0f;
     ai_cache.my_weapon_value = weapon_value( weapon );
+    ai_cache.live_explosives = find_live_explosives( explosion_avoidance_radius );
```

Comment:
I'd keep it to friendly NPCs. Consider the following exploit:
* Set C-4 to a year
* Set it by your door
* Go to sleep knowing that no NPC will dare to get near

Reply:
NPCs are now aware of explosion delays and ranges. They would simply ignore the armed explosives that aren't going to explode any time soon. Once it becomes a problem, they sprint away and keep safe distance (which is power-dependent).

Reply:
Sprint? As in actually run faster than walk or just a figure of speech?
Last time I checked, panicked NPCs calmly walked away from things.

Reply:
You're right, of course: that was a figure of speech. They don't literally run. I suppose, this should be changed at some point.